sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -188,7 +193,13 @@ class DefaultTimestampFormatter(
 
   override def parseWithoutTimeZone(s: String): Long = {
     try {
-      DateTimeUtils.stringToTimestampWithoutTimeZoneAnsi(UTF8String.fromString(s))
+      val utf8Value = UTF8String.fromString(s)
+      val (_, zoneIdOpt, _) = DateTimeUtils.parseTimestampString(utf8Value)
```

Comment:
This method will be run twice now. When do we use `DefaultTimestampFormatter`? If it's used rarely then the perf does not matter.

Reply:
No, this is the default timestamp formatter. If a user does not provide "timestampFormat", they will use DefaultTimestampFormatter.

Reply:
I can take a look to see if we can avoid this double calls.

Reply:
e.g. we can add an overload of `stringToTimestampWithoutTimeZoneAnsi`, to take an extra bool flag for failing with timezone